jscomp/runtime/caml_obj.res

```diff
@@ -303,53 +299,46 @@ type eq = (Obj.t, Obj.t) => bool
     basic type is not the same, it will not equal 
 */
 let rec equal = (a: Obj.t, b: Obj.t): bool =>
-  /* front and formoest, we do not compare function values */
   if a === b {
     true
   } else {
     let a_type = Js.typeof(a)
-    if (
-      a_type == "string" ||
-        (a_type == "number" ||
-        (a_type == "bigint" ||
-          (a_type == "boolean" ||
-          (a_type == "undefined" || a === %raw(`null`)))))
-    ) {
+    if a_type !== "object" || a === %raw(`null`) {
```

Comment:
The same, but shorter, faster, easier to maintain, and prevents from forgetting symbol and other types.